pkg/resources/view_grant.go

```diff
@@ -50,8 +51,16 @@ var viewGrantSchema = map[string]*schema.Schema{
 		Type:        schema.TypeSet,
 		Elem:        &schema.Schema{Type: schema.TypeString},
 		Optional:    true,
-		Description: "Grants privilege to these shares.",
+		Description: "Grants privilege to these shares (only valid if future_views = false).",
+		ForceNew:    true,
+	},
+	"future_views": &schema.Schema{
```

Comment:
I think we should name this "on_future" so that it maps more clearly to what SQL will be run.

Reply:
The other option I was considering was to not add a bool field, and just let an empty view_name imply future views.  Just a thought, and isn't too hard to switch back and forth, but we need to come to a decision before merging and releasing this.

Reply:
I would worry a bit about that implicit semantic. I like having the boolean to make it more clear what is happening.